src/toil/jobStores/aws/jobStore.py

```diff
@@ -1153,39 +1185,122 @@ def __getBucketVersioning(self, bucket):
         for attempt in retry_s3():
             with attempt:
                 status = bucket.get_versioning_status()
-        return bool(status) and self.versionings[status['Versioning']]
+        return bool(status) and cls.versionings[status['Versioning']]
 
-    def __getBucketRegion(self, bucket):
+    @staticmethod
+    def __getBucketRegion(bucket):
         for attempt in retry_s3():
             with attempt:
                 return bucket_location_to_region(bucket.get_location())
 
     def deleteJobStore(self):
-        self.registry_domain.put_attributes(self.namePrefix, dict(exists=str(False)))
-        if self.filesBucket is not None:
-            for attempt in retry_s3():
-                with attempt:
-                    for upload in self.filesBucket.list_multipart_uploads():
-                        upload.cancel_upload()
-            if self.__getBucketVersioning(self.filesBucket) in (True, None):
+        self.cleanJobStore('aws:%s:%s' % (self.region, self.namePrefix))
+
+    @classmethod
+    def cleanJobStore(cls, locator):
+        info = cls._processLocator(locator)
+        filesBucket = None
+
+        # First we load all job store components.
+        try:
+            with closing(cls._connectS3(info.region)) as s3:
                 for attempt in retry_s3():
                     with attempt:
-                        for key in list(self.filesBucket.list_versions()):
-                            self.filesBucket.delete_key(key.name, version_id=key.version_id)
+                        filesBucket = s3.get_bucket('%s%sfiles' %
+                                                    (info.namePrefix, cls.nameSeparator))
+        except S3ResponseError as e:
+            if e.error_code == 'NoSuchBucket':
+                pass
             else:
-                for attempt in retry_s3():
-                    with attempt:
-                        for key in list(self.filesBucket.list()):
-                            key.delete()
-            for attempt in retry_s3():
-                with attempt:
-                    self.filesBucket.delete()
-        for domain in (self.filesDomain, self.jobsDomain):
-            if domain is not None:
-                for attempt in retry_sdb():
-                    with attempt:
-                        domain.delete()
+                raise
+
+        def getDomain(name):
+            # no such domain predicate has been removed to prevent unnecessary retries
+            predicate = (lambda e: sdb_unavailable(e) or
```

Comment:
Compose the predicate without duplicating the existing one. If the default one is

```
def p1():
     return a or b or c or d
```

make the new one 

```
def p2():
    return not d and p1
```